src/utils.js

```diff
@@ -0,0 +1,69 @@
+var _ = require('lodash');
+
+function format_address(address) {
+    if (isNoU(address) || address.length === 0) {
+        return '';
+    }
+
+    var len = address.length;
+    var working_len = 0;
+
+    if (len > 4) {
+        working_len = len - 4;
+    }
+
+    var rep_str = '';
+    var retVal = '';
+
+    if (working_len > 0) {
+        var rx = new RegExp('.{' + working_len + '}(.*)');            
+
+        rep_str = Array(working_len + 1).join('*');
+        retVal = address.replace(rx, rep_str + '$1');
+    }
+    else {
+        retVal = address;
+    }
+
+    return retVal;
+}
+
+function send_email(email_config, usr_addr) {
+}
+
+function process_keywords(msg, keywords, email_config) {
+    var usr_msg = msg.content.toLowerCase();
+    var usr_addr = format_address(msg.from_addr);
+
+    if (
+        isNoU(usr_msg) || 
+        isNoU(keywords) || 
+        !_.isArray(keywords) || 
+        (_.isArray(keywords) && keywords.length === 0)
+    ) {
+        return;
+    }
+
+    _.each( keywords, function(keyword) {
+        
```

Comment:
Guessing this is still being written, which is fine for now. Just noting it in case it wasn't.

Note that you probably don't want to call `send_email` at the end of this function (you should instead return a list of matching keyword). This will become more obvious when you implement `send_email` which will require returning a promise.